diff --git a/slider/control/SliderControl.mjs b/slider/control/SliderControl.mjs
index 9a98402635d582f77de353bdcd0727b23509c5dd..adb1aa0314882163ba356082e9133dfc4f6c22a9 100644
--- a/slider/control/SliderControl.mjs
+++ b/slider/control/SliderControl.mjs
@@ -316,13 +316,15 @@ export const SliderControl = /*#__PURE__*/React.forwardRef(function SliderContro
       x: touch.clientX,
       y: touch.clientY
     };
-    startPressing(fingerCoords);
-    const finger = getFingerState(fingerCoords);
-    if (finger == null) {
-      return;
+    if (currentInteractionValueRef.current == null) {
+      startPressing(fingerCoords);
+      const finger = getFingerState(fingerCoords);
+      if (finger == null) {
+        return;
+      }
+      focusThumb(finger.thumbIndex);
+      setValueFromPointer(finger, REASONS.trackPress, nativeEvent);
     }
-    focusThumb(finger.thumbIndex);
-    setValueFromPointer(finger, REASONS.trackPress, nativeEvent);
     moveCountRef.current = 0;
     const doc = ownerDocument(controlRef.current);
     doc.addEventListener('touchmove', handleTouchMove, {
@@ -331,13 +333,18 @@ export const SliderControl = /*#__PURE__*/React.forwardRef(function SliderContro
     doc.addEventListener('touchend', handleTouchEnd, {
       passive: true
     });
+    doc.addEventListener('touchcancel', handleTouchEnd, {
+      passive: true
+    });
   });
   const stopListening = useStableCallback(() => {
     const doc = ownerDocument(controlRef.current);
     doc.removeEventListener('pointermove', handleTouchMove);
     doc.removeEventListener('pointerup', handleTouchEnd);
+    doc.removeEventListener('pointercancel', handleTouchEnd);
     doc.removeEventListener('touchmove', handleTouchMove);
     doc.removeEventListener('touchend', handleTouchEnd);
+    doc.removeEventListener('touchcancel', handleTouchEnd);
     pressedValuesRef.current = null;
     currentInteractionValueRef.current = null;
   });
@@ -411,6 +418,9 @@ export const SliderControl = /*#__PURE__*/React.forwardRef(function SliderContro
         doc.addEventListener('pointerup', handleTouchEnd, {
           once: true
         });
+        doc.addEventListener('pointercancel', handleTouchEnd, {
+          once: true
+        });
       }
     }, elementProps],
     stateAttributesMapping: sliderStateAttributesMapping
