diff --git a/dist/vanilla-extract-css.esm.js b/dist/vanilla-extract-css.esm.js
index 37e8e83bc47dc12d761f9347cfa3cd7bba15814b..6385831bff191c9c48ceec274cbda5b09f3ff8bb 100644
--- a/dist/vanilla-extract-css.esm.js
+++ b/dist/vanilla-extract-css.esm.js
@@ -126,7 +126,16 @@ function getDevPrefix(_ref) {
   return parts.join('_');
 }
 function normalizeIdentifier(identifier) {
-  return identifier.match(/^[0-9]/) ? "_".concat(identifier) : identifier;
+  return identifier.match(/^[A-Za-z_]/) ? identifier : "_".concat(identifier);
+}
+function toBase64(value) {
+  var alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";
+  var result = "";
+  do {
+    result = alphabet[value % 64] + result;
+    value = Math.floor(value / 64);
+  } while (value > 0);
+  return result;
 }
 function generateIdentifier(arg) {
   var identOption = getIdentOption();
@@ -137,13 +146,12 @@ function generateIdentifier(arg) {
     debugId: arg
   } : null), typeof arg === 'object' ? arg : null);
 
-  // Convert ref count to base 36 for optimal hash lengths
-  var refCount = getAndIncrementRefCounter().toString(36);
+  var refCount = toBase64(getAndIncrementRefCounter());
   var {
     filePath,
     packageName
   } = getFileScope();
-  var fileScopeHash = hash(packageName ? "".concat(packageName).concat(filePath) : filePath);
+  var fileScopeHash = toBase64(parseInt(hash(packageName ? "".concat(packageName).concat(filePath) : filePath), 36));
   var identifier = "".concat(fileScopeHash).concat(refCount);
   if (identOption === 'debug') {
     var devPrefix = getDevPrefix({
