diff --git a/dist/doc.d.ts b/dist/doc.d.ts
index ef9a7f2035c14c5d4a8b2ec42117bf31ca6b00b7..d00a26ed0200e49e501d3e0e7adddca8ba416c6f 100644
--- a/dist/doc.d.ts
+++ b/dist/doc.d.ts
@@ -1015,13 +1015,13 @@ declare namespace Pos {
         */
         get isLast(): boolean;
         /**
-        The node before this one, if any.
+        The node after this one, if any.
         */
-        get nextSibling(): Node | null;
+        get nextSibling(): _Node | null;
         /**
-        The node after this node, if any.
+        The node before this one, if any.
         */
-        get previousSibling(): Node | null;
+        get previousSibling(): _Node | null;
     }
     /**
     Subclass of {@link Pos.Node} that points at a plot node.
diff --git a/dist/editor.js b/dist/editor.js
index 04df810151c72f62dafa53f7f19e8d66853a55fb..fc620aec5bf69f4ea08c94a8aaeebd7841fa6d1e 100644
--- a/dist/editor.js
+++ b/dist/editor.js
@@ -669,7 +669,7 @@ class PointIterator {
     goto(pos, inclusive) {
         let i = findAbove(this.set.positions, 0, pos - 1);
         if (!inclusive) {
-            while (i < this.set.values.length && this.set.values[i].side < 1000000000)
+            while (i < this.set.values.length && this.set.positions[i] == pos && this.set.values[i].side < 1000000000)
                 i++;
         }
         this.fill(i);
@@ -2529,18 +2529,19 @@ class ContentUpdate {
                     index++;
                 }
             }
-        for (let i = cut.length; i;) {
-            let from = cut[--i], to = Math.min(from + 1, includeEnd ? end : end - 1);
-            while (i && cut[i - 1] == from - 1) {
-                from--;
-                i--;
-            }
-            if (from > this.posB)
+        for (let i = 0; i < cut.length;) {
+            let from = cut[i++], last = from;
+            while (i < cut.length && cut[i] == last + 1)
+                last = cut[i++];
+            let to = Math.min(last + 1, includeEnd ? end : end - 1);
+            if (from > this.posB) {
                 this.keepInner(from - this.posB, includeStart, false);
-            this.update(to - from, true);
+                includeStart = true;
+            }
+            this.update(to - from, includeStart);
             includeStart = false;
             len = end - to;
         }
         if (len)
             this.keepInner(len, includeStart, includeEnd);
     }
diff --git a/dist/state.js b/dist/state.js
index 0d110081825754f6adaf5368d676794b8d563a82..2fb8b613dd0724bc83b097e76897515fc8cccf40 100644
--- a/dist/state.js
+++ b/dist/state.js
@@ -1680,7 +1680,7 @@ class GardState {
         }
     });
 ;return GardState})(GardState);
-const initField = /*@__PURE__*/GardState.Facet.define({ static: true });
+const initField = GardState.Facet.define({ static: true });
 function addValue(set, value) {
     if (set.indexOf(value) < 0)
         set.push(value);
